cdap-security/src/main/java/io/cdap/cdap/security/auth/context/MasterAuthenticationContext.java

```diff
@@ -37,13 +44,33 @@
  * @see UserGroupInformation
  */
 public class MasterAuthenticationContext implements AuthenticationContext {
+  private final SConfiguration sConf;
+  private final AESCipher aesCipher;
+
+  @Inject
+  public MasterAuthenticationContext(SConfiguration sConf) {
+    this.sConf = sConf;
+    this.aesCipher = new AESCipher();
+  }
 
   @Override
   public Principal getPrincipal() {
     // When requests come in via rest endpoints, the userId is updated inside SecurityRequestContext, so give that
     // precedence.
     String userId = SecurityRequestContext.getUserId();
-    String userCredential = SecurityRequestContext.getUserCredential();
+    String userCredential = SecurityRequestContext.getUserCredential(); //
```

Comment:
Discussed offline, it might be good to move the decryption to the DefaultAccessEnforcer class to ensure compatibility between this and internal identity.